### PR TITLE
Implement org.freedesktop.impl.portal.Email

### PR DESCRIPTION
**Pull request by matthiasclasen**
This is a very simple implmentation that just calls out
to evolution.

### CODE REVIEW

**Comment by mariospr** [comment]
Typo in the commit msg: implmentation -> implementation

File: src/xdg-desktop-portal-gtk.c [hunk 1]
```diff
@@ -130,6 +131,12 @@ on_bus_acquired (GDBusConnection *connection,
       g_warning ("error: %s\n", error->message);
       g_clear_error (&error);
     }
+
+  if (!email_init (connection, &error))
+    {
+      g_warning ("error: %s\n", error->message);
+      g_clear_error (&error);
+    }
```
Comment (mariospr) [inline src/xdg-desktop-portal-gtk.c]: Similar comment than https://github.com/flatpak/xdg-desktop-portal/pull/86/commits/32550005522e38b5bba98b26d10dc5ff455af281#r101017048: not for now, but looks like this function could use some refactoring too

File: src/email.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
+#define _GNU_SOURCE 1
```
Comment (mariospr) [inline src/email.c]: Missing license header

**Comment by mariospr** [comment]
I placed a couple of comments in the PR, but it looks good to me in general. 

This said, I think it would be nice to support more clients (at least Thunderbird), but that sounds like it would need extra code (https://github.com/flatpak/xdg-desktop-portal/pull/86#issuecomment-279370754), and not to hardcode the path to the evolution binary if possible, but I think both things can be achieved in separate commit.

**Comment by mariospr** [comment]
Oh! And we'd need to change the proxy functions to match the suggested name of ComposeEmail too, if we do that change in https://github.com/flatpak/xdg-desktop-portal/pull/86, of course.

File: src/email.c [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  g_string_append_printf (url, "?subject=%s", enc_subject);
+  g_string_append_printf (url, "&body=%s", enc_body);
+
+  for (i = 0; attachments[i]; i++)
+    {
+      g_autofree char *path = g_uri_escape_string (attachments[i], NULL, FALSE);
+      g_string_append_printf (url, "&attach=%s", path);
+    }
+
+  argv[0] = "evolution";
```
Comment (AsciiWolf) [inline src/email.c]: I think a better idea would be to use something like _xdg-email_ instead of using hardcoded email client(s).

Comment (matthiasclasen) [inline src/email.c]: xdg-email is a horrible shell script. I'm not going to use it

Comment (AsciiWolf) [inline src/email.c]: What about _xdg-open_? I think it also has email handling support.

Comment (matthiasclasen) [inline src/email.c]: Same comment

Comment (AsciiWolf) [inline src/email.c]: What about the new [flatpak-xdg-utils](https://github.com/flatpak/flatpak-xdg-utils)?

Comment (mariospr) [inline src/email.c]: The new flatpak-xdg-utils is meant to be run from inside the sandbox, as it relies on the portal (this one, actually) to open the email client. The code here is run in the host system, that's why it opens the application directly or something like xdg-email, but the one currently available is too bad, as @matthiasclasen said before.

**Pull request merged**